port/linux/omrvmem.c

```diff
@@ -1174,13 +1180,33 @@ reserve_memory_with_mmap(struct OMRPortLibrary *portLibrary, void *address, uint
 			useBackingSharedFile = FALSE;
 			Trc_PRT_vmem_reserve_doubleMapAPI_not_available(address, byteAmount);
 		}
+	} else if (useBackingSharedTmpFile) {
+		/* Generate a unique temporary filename from template and open the file. */
+		char filename[FILE_NAME_SIZE + 1];
+		snprintf(filename, sizeof(filename), "omrvmem_%09d_XXXXXX", getpid());
+		fd = mkostemp(filename, 0);
+		if (OMRPORT_INVALID_FD != fd) {
+			unlink(filename);
+			/* Set the file size with ftruncate. */
+			if (OMRPORT_INVALID_FD == ftruncate(fd, byteAmount)) {
+				close(fd);
+				fd = OMRPORT_INVALID_FD;
+			}
+		}
+		if (OMRPORT_INVALID_FD == fd) {
+			/* Revert to using an ANONYMOUS mmap. */
+			useBackingSharedTmpFile = FALSE;
+			flags = 0;
+			useBackingFile = set_flags_for_mmap(&flags);
+			Trc_PRT_vmem_reserve_tempfile_not_created(filename, byteAmount);
+		}
 	} else if (useBackingFile) {
 		fd = portLibrary->file_open(portLibrary, "/dev/zero", EsOpenRead | EsOpenWrite, 0);
 	}
 
-	if (((OMRPORT_INVALID_FD == fd) && (!useBackingSharedFile && !useBackingFile)) ||
-	   ((OMRPORT_INVALID_FD != fd) && (useBackingSharedFile || useBackingFile))) {
-
+	if (((OMRPORT_INVALID_FD == fd) && (!useBackingSharedFile && !useBackingSharedTmpFile && !useBackingFile))
+	|| ((OMRPORT_INVALID_FD != fd) && (useBackingSharedFile || useBackingSharedTmpFile || useBackingFile))
+	) {
```

Comment:
I'm not a fan of this sort of condition which is effectively `(a && b) || (!a && !b)` but hard to recognize.
Perhaps with the help of suitably named local variables it should be expressed as `a == b`.

Reply:
In the latest push I have changed the code to:
```
	BOOLEAN backingFileNeeded = useBackingFile || useBackingSharedFile || useBackingSharedTmpFile;
	if ((OMRPORT_INVALID_FD != fd) || !backingFileNeeded) {
```
This should be equivalent to the original, given that there is no way you can have a valid `fd` if you don't need one (`backingFileNeeded`==`FALSE`)